src/aurora/erfwriter.cpp

```diff
@@ -0,0 +1,108 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Writing BioWare's ERFs (encapsulated resource file).
+ */
+
+#include <ctime>
+#include "erfwriter.h"
+
+namespace Aurora {
+
+ERFWriter::ERFWriter(uint32 id, uint32 fileCount, Common::SeekableWriteStream &stream, LocString description) :
+		_stream(stream), _id(id), _currentFileCount(0), _fileCount(fileCount), _description(description) {
+	stream.writeUint32BE(_id);
+	stream.writeUint32BE(MKTAG('V', '1', '.', '0'));
+
+	// Write Header.
+	stream.writeUint32LE(_description.getNumStrings()); // Language count.
+	stream.writeUint32LE(_description.getWrittenSize()); // Localized string size.
+
+	stream.writeUint32LE(_fileCount); // Entry Count.
+
+	uint32 locStringTableOffset = 160; // 160 is the size of the erf header.
+	_keyTableOffset = locStringTableOffset + _description.getWrittenSize();
+	_resourceTableOffset = _keyTableOffset + _fileCount * 24;
+
+	stream.writeUint32LE(locStringTableOffset); // LocString offset.
+	stream.writeUint32LE(_keyTableOffset); // Key List offset.
+	stream.writeUint32LE(_resourceTableOffset); // Resource offset.
+
+	// Write the creation time of the file.
+	std::time_t now = std::time(0);
+	std::tm *timepoint = std::localtime(&now);
+	stream.writeUint32LE(timepoint->tm_year);
+	stream.writeUint32LE(timepoint->tm_yday);
+
+	// Write the description string reference.
+	if (_description.getNumStrings())
+		stream.writeUint32LE(_description.getID());
+	else
+		stream.writeUint32LE(0);
+
+	// Write 116 bytes of reserved header data.
+	stream.writeZeros(116);
+
+	// Write the Localized string table.
+	_description.writeLocString(stream);
```

Comment:
Can you add a test for writing a LocString with multiple languages, please?

In fact, you should probably add a test for writing LocStrings to the LocStrings unit tests as well.